packages/react-components/react-avatar/src/components/AvatarGroupItem/useAvatarGroupItemStyles.ts

```diff
@@ -150,16 +155,57 @@ const useSpreadStyles = makeStyles({
   xl: { '&:not(:first-child)': { marginLeft: tokens.spacingHorizontalXL } },
 });
 
+/**
+ * Styles for the pie layout
+ */
+const usePieStyles = makeStyles({
+  base: {
+    position: 'absolute',
+  },
+  slices: {
+    // Two slices
+    '&:nth-of-type(1):nth-last-of-type(2)': {
+      clipPath: `inset(0 calc(25% + (var(${avatarGroupItemDividerWidthVar}) / 2)) 0 25%)`,
+      left: '-25%',
+    },
+    '&:nth-of-type(2):nth-last-of-type(1)': {
+      clipPath: `inset(0 25% 0 calc(25% + (var(${avatarGroupItemDividerWidthVar}) / 2)))`,
+      left: '25%',
+    },
+
+    // Three slices
+    '&:nth-of-type(1):nth-last-of-type(3)': {
+      clipPath: `inset(0 calc(25% + (var(${avatarGroupItemDividerWidthVar}) / 2)) 0 25%)`,
+      left: '-25%',
+    },
+    '&:nth-of-type(2):nth-last-of-type(2),&:nth-of-type(3):nth-last-of-type(1)': {
+      // Since the two AvatarGroupItems on the right are scaled by 0.5, the divider width should not be halved.
+      clipPath: `inset(0 0 var(${avatarGroupItemDividerWidthVar}) var(${avatarGroupItemDividerWidthVar}))`,
+      left: '50%',
+      transform: 'scale(0.5)',
+      transformOrigin: '0 0',
+    },
+    '&:nth-of-type(3):nth-last-of-type(1)': {
+      clipPath: `inset(var(${avatarGroupItemDividerWidthVar}) 0 0 var(${avatarGroupItemDividerWidthVar}))`,
+      top: '50%',
+    },
```

Comment:
(nit) Given that these styles are already quite complicated, it might make them easier to understand if you write out all of the styles for the third item, instead of grouping part of them with the second item and then overriding them later. It's more verbose, but IMO easier to understand.
```suggestion
    '&:nth-of-type(2):nth-last-of-type(2)': {
      // Since the two AvatarGroupItems on the right are scaled by 0.5, the divider width should not be halved.
      clipPath: `inset(0 0 var(${avatarGroupItemDividerWidthVar}) var(${avatarGroupItemDividerWidthVar}))`,
      left: '50%',
      transform: 'scale(0.5)',
      transformOrigin: '0 0',
    },
    '&:nth-of-type(3):nth-last-of-type(1)': {
      clipPath: `inset(var(${avatarGroupItemDividerWidthVar}) 0 0 var(${avatarGroupItemDividerWidthVar}))`,
      left: '50%',
      top: '50%',
      transform: 'scale(0.5)',
      transformOrigin: '0 0',
    },
```